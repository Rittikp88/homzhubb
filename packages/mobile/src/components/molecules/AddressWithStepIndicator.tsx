import React from 'react';
import { FlatList, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Badge, Label, Text } from '@homzhub/common/src/components';
import { PropertyAddressCountry } from '@homzhub/mobile/src/components/molecules/PropertyAddressCountry';
import { TypeOfPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { ILabelColor } from '@homzhub/common/src/domain/models/LeaseTransaction';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps {
  steps: string[];
  currentIndex: number;
  isStepDone: boolean[];
  primaryAddress: string;
  subAddress: string;
  propertyType: string;
  icon?: string;
  onEditPress?: () => void;
  selectedPan?: string;
  onPressSteps: (index: number) => void;
  badgeStyle?: StyleProp<ViewStyle>;
  stepContainerStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithStepIndicator = (props: IProps): React.ReactElement => {
  const {
    steps,
    currentIndex,
    isStepDone,
    icon,
    selectedPan,
    badgeStyle,
    stepContainerStyle = {},
    containerStyle = {},
    primaryAddress,
    subAddress,
    propertyType,
    onPressSteps,
    onEditPress,
  } = props;

  const { t } = useTranslation(LocaleConstants.namespacesKey.property);

  const { rental, sell, informational, blue, green, darkTint7 } = theme.colors;

  const badgeData = (): ILabelColor => {
    switch (selectedPan) {
      case TypeOfPlan.RENT:
        return {
          label: t('forRental'),
          color: rental,
        };
      case TypeOfPlan.SELL:
        return {
          label: t('forSale'),
          color: sell,
        };
      case TypeOfPlan.MANAGE:
        return {
          label: t('manage'),
          color: informational,
        };
      default:
        return {
          label: t('forRental'),
          color: rental,
        };
    }
  };

  const renderIndicator = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const isCurrentStep = currentIndex === index;
    const stepDone = isStepDone[index];
    const iconColor = isCurrentStep ? blue : stepDone ? green : darkTint7;
    const labelStyle = [styles.stepLabel, isCurrentStep && styles.currentStepLabel, stepDone && styles.doneStepLabel];
    return (
      <View style={styles.indicatorView}>
        <Icon name={icons.roundFilled} size={30} color={iconColor} onPress={(): void => onPressSteps(index)} />
        <Label type="small" textType="semiBold" style={labelStyle}>
          {item}
        </Label>
      </View>
    );
  };

  const renderSeparator = (separatorData: { highlighted: boolean; leadingItem: string }): React.ReactElement => {
    let isDone = false;
    const indicatorSteps: string[] = steps;
    indicatorSteps.forEach((item, index) => {
      if (separatorData.leadingItem === item) {
        isDone = isStepDone[index];
      }
    });
    return <View style={[styles.separator, isDone && styles.doneSeparator]} />;
  };

  const badge = badgeData();

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.topView}>
        <Text type="small" textType="regular">
          {propertyType}
        </Text>
        {icon && <Icon name={icon} size={23} color={blue} onPress={onEditPress} />}
        {selectedPan && <Badge title={badge.label.toUpperCase()} badgeColor={badge.color} badgeStyle={badgeStyle} />}
      </View>
      <PropertyAddressCountry
        primaryAddress={primaryAddress}
        subAddress={subAddress}
        containerStyle={styles.addressView}
      />
      <FlatList
        keyExtractor={(step): string => step}
        data={steps}
        renderItem={renderIndicator}
        horizontal
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={renderSeparator}
        style={[styles.listStyle, stepContainerStyle]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  addressView: {
    marginVertical: 6,
  },
  indicatorView: {
    alignItems: 'center',
  },
  topView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  separator: {
    backgroundColor: theme.colors.darkTint7,
    width: 50,
    height: 2,
    marginTop: 10,
    opacity: 0.3,
  },
  doneSeparator: {
    backgroundColor: theme.colors.green,
  },
  listStyle: {
    marginTop: 10,
    alignSelf: 'center',
  },
  stepLabel: {
    color: theme.colors.darkTint7,
  },
  currentStepLabel: {
    color: theme.colors.blue,
  },
  doneStepLabel: {
    color: theme.colors.green,
  },
});
