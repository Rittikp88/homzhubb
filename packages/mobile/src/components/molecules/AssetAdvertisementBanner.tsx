import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { AssetAdvertisement, AssetAdvertisementResults } from '@homzhub/common/src/domain/models/AssetAdvertisement';

interface IAssetAdvertisementBannerState {
  banners: AssetAdvertisement;
  activeSlide: number;
}

export class AssetAdvertisementBanner extends React.PureComponent<{}, IAssetAdvertisementBannerState> {
  public state = {
    banners: {} as AssetAdvertisement,
    activeSlide: 0,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchAdvertisementBanners();
  };

  public render(): React.ReactNode {
    const { banners, activeSlide } = this.state;
    if (!banners) {
      return null;
    }
    return (
      <View style={styles.container}>
        <SnapCarousel
          carouselData={banners?.results ?? []}
          carouselItem={this.renderCarouselItem}
          itemWidth={theme.viewport.width}
          activeIndex={activeSlide}
          onSnapToItem={this.onSnapToItem}
          testID="bannerSnap"
        />
        <PaginationComponent
          dotsLength={banners?.results?.length ?? 0}
          activeSlide={activeSlide}
          containerStyle={styles.overlay}
          activeDotStyle={styles.activeDotStyle}
          inactiveDotStyle={styles.inactiveDotStyle}
        />
      </View>
    );
  }

  private renderCarouselItem = (item: AssetAdvertisementResults): React.ReactElement => {
    return (
      <Image
        source={{
          uri: item.attachment.link,
        }}
        style={styles.carouselImage}
      />
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  public fetchAdvertisementBanners = async (): Promise<void> => {
    const response: AssetAdvertisement = await DashboardRepository.getAdvertisements();
    this.setState({ banners: response });
  };
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    height: 180,
    overflow: 'hidden',
  },
  overlay: {
    position: 'absolute',
    flex: 0,
    alignSelf: 'center',
    bottom: 0,
    paddingVertical: 5,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
  activeDotStyle: {
    borderWidth: 1,
  },
  inactiveDotStyle: {
    backgroundColor: theme.colors.darkTint6,
    borderColor: theme.colors.transparent,
  },
});
