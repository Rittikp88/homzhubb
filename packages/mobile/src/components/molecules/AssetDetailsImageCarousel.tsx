import React from 'react';
import { View, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { ImageVideoPagination } from '@homzhub/common/src/components';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';

interface IProps {
  images: any;
  enterFullScreen: () => void;
  updateSlide: (index: number) => void;
  activeSlide: number;
}

export class AssetDetailsImageCarousel extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { activeSlide, images } = this.props;
    return (
      <View style={styles.carouselContainer}>
        <SnapCarousel
          carouselData={images}
          carouselItem={this.renderCarouselItem}
          activeIndex={activeSlide}
          sliderWidth={theme.viewport.width}
          itemWidth={theme.viewport.width}
          onSnapToItem={this.onSnapToItem}
        />
        <View style={styles.overlay}>
          <ImageVideoPagination
            currentSlide={activeSlide}
            totalSlides={images.length}
            type={images[activeSlide].attachment_type}
          />
        </View>
      </View>
    );
  }

  private renderCarouselItem = (item: any): React.ReactElement => {
    const { enterFullScreen } = this.props;
    return (
      <TouchableOpacity onPress={enterFullScreen}>
        {item.attachment_type === 'IMAGE' && (
          <Image
            source={{
              uri: item.link,
            }}
            style={styles.carouselImage}
          />
        )}
        {/* TODO: Remove once the api is integrated */}
        {item.attachment_type === 'VIDEO' && <Image source={item.thumbnail_url} style={styles.carouselImage} />}
      </TouchableOpacity>
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    const { updateSlide } = this.props;
    updateSlide(slideNumber);
  };
}

const styles = StyleSheet.create({
  carouselContainer: {
    borderRadius: 4,
    overflow: 'hidden',
    backgroundColor: theme.colors.darkTint1,
  },
  overlay: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 15,
    backgroundColor: theme.colors.carouselCardOpacity,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
});
