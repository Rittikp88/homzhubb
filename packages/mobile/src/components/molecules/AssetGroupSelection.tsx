import React from 'react';
import { View, StyleSheet, TouchableOpacity, ScrollView } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon from '@homzhub/common/src/assets/icon';
import { Label, Text } from '@homzhub/common/src/components';
import { AssetGroup } from '@homzhub/common/src/domain/models/AssetGroup';
import { Unit } from '@homzhub/common/src/domain/models/Unit';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps extends WithTranslation {
  assetGroups: AssetGroup[];
  selectedAssetGroupType: number;
  selectedAssetGroupId: number;
  scrollRef: ScrollView | null;
  onAssetGroupSelected: (id: number) => void;
  isDisabled: boolean;
}

interface IOwnState {
  selectedAssetGroup: number;
}

enum Type {
  // eslint-disable-next-line no-shadow
  AssetGroup = 'AssetGroup',
  AssetType = 'AssetType',
}

class AssetGroupSelection extends React.PureComponent<IProps, IOwnState> {
  public state = {
    selectedAssetGroup: -1,
  };

  public static getDerivedStateFromProps(props: IProps, state: IOwnState): IOwnState | null {
    const { selectedAssetGroupId: newPropValue } = props;
    const { selectedAssetGroup: oldStateValue } = state;

    if (oldStateValue !== newPropValue && newPropValue !== -1) {
      return {
        selectedAssetGroup: newPropValue,
      };
    }

    return null;
  }

  public render = (): React.ReactNode => {
    const { assetGroups, t, isDisabled } = this.props;
    const { selectedAssetGroup } = this.state;

    const selectedGroup = assetGroups.find((assetGroup: AssetGroup) => assetGroup.id === selectedAssetGroup);

    let opacity;
    let pointer;
    if (isDisabled) {
      opacity = 0.5;
      pointer = 'none';
    }

    return (
      <View style={[styles.container, { opacity }]} pointerEvents={pointer as 'none' | undefined}>
        <Text type="small" textType="semiBold" style={styles.title}>
          {t('myProjectIs')}
        </Text>
        <View style={styles.mainGroupContainer}>
          {assetGroups.map((group, index) => this.renderItem(Type.AssetGroup, group, index))}
        </View>
        {selectedAssetGroup >= 0 && (
          <>
            <Text type="small" textType="semiBold" style={styles.title}>
              {t('selectType')}
            </Text>
            <View style={styles.subGroupContainer}>
              {selectedGroup?.assetTypes.map((item, index) => this.renderItem(Type.AssetType, item, index))}
            </View>
          </>
        )}
      </View>
    );
  };

  private renderItem = (type: Type, item: AssetGroup | Unit, index: number): React.ReactNode => {
    const { selectedAssetGroup } = this.state;
    const { onAssetGroupSelected, selectedAssetGroupType, scrollRef } = this.props;

    const conditionalSelectedItem = type === Type.AssetGroup ? selectedAssetGroup : selectedAssetGroupType;
    const conditionalContainerStyle =
      type === Type.AssetGroup
        ? styles.itemContainer
        : [styles.subItemContainer, { marginEnd: index % 2 === 0 ? marginEnd : 0 }];

    let color = theme.colors.darkTint4;
    let backgroundColor;
    let textType = 'regular';
    if (item.id === conditionalSelectedItem) {
      color = theme.colors.white;
      backgroundColor = theme.colors.primaryColor;
      textType = 'semiBold';
    }

    const onPress = (): void => {
      if (type === Type.AssetGroup) {
        this.setState({ selectedAssetGroup: item.id });
        onAssetGroupSelected(-1);
        setTimeout(() => {
          scrollRef?.scrollToEnd({ animated: true });
        }, 0);
      } else {
        onAssetGroupSelected(item.id);
      }
    };

    return (
      <TouchableOpacity key={item.id} onPress={onPress} style={[conditionalContainerStyle, { backgroundColor }]}>
        {type === Type.AssetGroup && <Icon name={item.name.toLowerCase()} size={24} color={color} />}
        <Label type="large" textType={textType as 'regular' | 'semiBold'} style={[styles.assetGroupName, { color }]}>
          {item.name}
        </Label>
      </TouchableOpacity>
    );
  };
}

const {
  layout: { screenPadding },
  viewport: { width },
} = theme;
const marginEnd = 12;

const styles = StyleSheet.create({
  container: {
    marginHorizontal: screenPadding,
    marginBottom: 12,
  },
  mainGroupContainer: {
    flexDirection: 'row',
  },
  subGroupContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  itemContainer: {
    alignItems: 'center',
    marginEnd: 12,
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
  },
  subItemContainer: {
    width: (width - (screenPadding * 2 + marginEnd)) / 2,
    paddingVertical: 4,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
  },
  assetGroupName: {
    textAlign: 'center',
    marginEnd: 4,
  },
  title: {
    textAlign: 'center',
    marginTop: 24,
    marginBottom: 16,
  },
});

const HOC = withTranslation(LocaleConstants.namespacesKey.property)(AssetGroupSelection);
export { HOC as AssetGroupSelection };
