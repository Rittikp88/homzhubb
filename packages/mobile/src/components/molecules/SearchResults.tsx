import React from 'react';
import { FlatList, StyleSheet, TouchableOpacity } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components';
import { GooglePlaceData } from '@homzhub/common/src/services/GooglePlaces/interfaces';

export interface IProps extends WithTranslation {
  results: GooglePlaceData[];
  onResultPress: (item: GooglePlaceData) => void;
}

class SearchResults extends React.PureComponent<IProps, {}> {
  public render = (): React.ReactNode => {
    const { results } = this.props;
    return (
      <FlatList
        keyboardShouldPersistTaps="always"
        data={results}
        renderItem={this.renderSearchResult}
        ListHeaderComponent={this.renderListHeader}
        contentContainerStyle={styles.contentContainer}
      />
    );
  };

  private renderListHeader = (): React.ReactElement => {
    const { t } = this.props;
    return (
      <Label type="large" textType="semiBold" style={styles.listTitle}>
        {t('common:searchResults')}
      </Label>
    );
  };

  private renderSearchResult = ({ item }: { item: GooglePlaceData }): React.ReactElement => {
    const { onResultPress } = this.props;

    const onPress = (): void => {
      onResultPress(item);
    };

    return (
      <TouchableOpacity style={styles.listItemContainer} onPress={onPress}>
        <Label type="large" style={styles.listItemTitle} numberOfLines={2}>
          {item.description}
        </Label>
      </TouchableOpacity>
    );
  };
}

const styles = StyleSheet.create({
  contentContainer: {
    backgroundColor: theme.colors.secondaryColor,
  },
  listItemContainer: {
    marginStart: 16,
    borderBottomWidth: 1,
    borderBottomColor: theme.colors.darkTint10,
  },
  listTitle: {
    backgroundColor: theme.colors.background,
    color: theme.colors.darkTint4,
    paddingTop: 20,
    paddingStart: 16,
    paddingBottom: 6,
  },
  listItemTitle: {
    marginVertical: 16,
    marginEnd: 16,
  },
});

const HOC = withTranslation()(SearchResults);
export { HOC as SearchResults };
