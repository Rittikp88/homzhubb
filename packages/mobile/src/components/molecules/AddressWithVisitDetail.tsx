import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { DateUtils } from '@homzhub/common/src/utils/DateUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { PropertyAddress } from '@homzhub/common/src/components/molecules/PropertyAddress';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { VisitSlot } from '@homzhub/common/src/mocks/BookVisit';

interface IProps {
  primaryAddress: string;
  subAddress: string;
  startDate: string;
  endDate: string;
  comments?: string;
  isMissedVisit?: boolean;
  isCompletedVisit?: boolean;
  isRescheduleAll?: boolean;
  isFromProperty?: boolean;
  onPressSchedule?: () => void;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithVisitDetail = (props: IProps): React.ReactElement => {
  const {
    subAddress,
    primaryAddress,
    startDate,
    comments,
    isMissedVisit,
    isCompletedVisit,
    containerStyle = {},
    onPressSchedule,
    isRescheduleAll = false,
    isFromProperty = false,
  } = props;
  const { t } = useTranslation(LocaleConstants.namespacesKey.property);
  const dateTime = DateUtils.convertTimeFormat(startDate, 'YYYY-MM-DD HH');
  const time = VisitSlot.find((item) => item.from === Number(dateTime[1]));
  const textStyle = [styles.textColor, isMissedVisit && styles.missedColor];
  return (
    <View style={containerStyle}>
      {isFromProperty ? (
        <>
          {!!comments && (
            <Text type="small" textType="semiBold" style={styles.addressContainer}>
              {comments}
            </Text>
          )}
        </>
      ) : (
        <PropertyAddress
          primaryTextType="small"
          subAddress={subAddress}
          primaryAddress={primaryAddress}
          subAddressStyle={styles.subAddress}
          containerStyle={styles.addressContainer}
        />
      )}
      <Label type="regular" style={[styles.textColor, !comments && styles.detail]}>
        {t('visitDetails')}
      </Label>
      <View style={styles.detailContainer}>
        <View style={styles.content}>
          <Text type="small" textType="semiBold" style={textStyle}>
            {DateUtils.getDisplayDate(dateTime[0], 'DD MMM')}
          </Text>
          <Icon
            name={icons.roundFilled}
            color={isMissedVisit ? theme.colors.error : theme.colors.darkTint3}
            size={8}
            style={styles.iconStyle}
          />
          <Text type="small" textType="semiBold" style={textStyle}>
            {time?.formatted}
          </Text>
        </View>
        <TouchableOpacity style={styles.content} onPress={onPressSchedule}>
          <Icon name={icons.schedule} color={theme.colors.blue} size={20} />
          <Text type="small" style={styles.scheduleText}>
            {isCompletedVisit ? t('newVisit') : isRescheduleAll ? t('rescheduleAll') : t('reschedule')}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  subAddress: {
    marginLeft: 0,
  },
  addressContainer: {
    marginVertical: 12,
  },
  textColor: {
    color: theme.colors.darkTint3,
  },
  detail: {
    marginTop: 12,
  },
  missedColor: {
    color: theme.colors.error,
  },
  detailContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconStyle: {
    marginTop: 6,
    marginHorizontal: 6,
  },
  scheduleText: {
    color: theme.colors.blue,
    marginStart: 6,
  },
});
