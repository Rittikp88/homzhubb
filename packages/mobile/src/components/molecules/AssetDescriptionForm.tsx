import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useTranslation } from 'react-i18next';
import { FormikProps, FormikValues } from 'formik';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import { FormCalendar, FormDropdown, FormTextInput, Text } from '@homzhub/common/src/components';
import { FormCounter } from '@homzhub/common/src/components/molecules/FormCounter';
import { AssetDescriptionDropdownValues } from '@homzhub/common/src/domain/models/AssetDescriptionForm';

interface IProps {
  formProps: FormikProps<FormikValues>;
  dropDownOptions: AssetDescriptionDropdownValues;
}

const AssetDescriptionForm = ({ formProps, dropDownOptions }: IProps): React.ReactElement => {
  const [t] = useTranslation(LocaleConstants.namespacesKey.property);
  // Todo (Sriram: Look into this prefill issue of carpetArea)

  return (
    <>
      <Text type="small" style={styles.headingStyle}>
        {t('assetDescription:description')}
      </Text>
      <View style={styles.formContainer}>
        <View style={styles.contentView}>
          <View style={styles.subContentView}>
            <FormTextInput
              style={styles.inputFieldStyle}
              name="carpetArea"
              label={t('carpetArea')}
              maxLength={10}
              numberOfLines={1}
              inputType="number"
              placeholder={t('common:enter')}
              formProps={formProps}
            />
          </View>
          <View style={styles.flexOne}>
            <FormDropdown
              name="areaUnit"
              label={t('areaUnit')}
              options={dropDownOptions.areaUnitDropdownValues}
              placeholder={t('selectAreaUnit')}
              maxLabelLength={8}
              formProps={formProps}
            />
          </View>
        </View>
        <FormDropdown
          label={t('facingText')}
          name="facing"
          options={dropDownOptions.facing}
          placeholder={t('propertySearch:selectFacing')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormDropdown
          label={t('flooringType')}
          name="flooringType"
          options={dropDownOptions.typeOfFlooring}
          placeholder={t('selectFlooringType')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormCalendar
          formProps={formProps}
          label={t('yearOfConstruction')}
          name="yearOfConstruction"
          allowPastDates
          isYearView
          placeHolder={t('assetFinancial:addDatePlaceholder')}
          placeHolderStyle={styles.placeHolderStyle}
          dateStyle={styles.dateStyle}
          textType="label"
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="totalFloors"
          maxCount={1000}
          label={t('numberOfFloorsText')}
          formProps={formProps}
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="onFloorNumber"
          label={t('onFloorText')}
          formProps={formProps}
          maxCount={formProps.values.totalFloors}
        />
      </View>
    </>
  );
};

const memoizedComponent = React.memo(AssetDescriptionForm);
export { memoizedComponent as AssetDescriptionForm };

const styles = StyleSheet.create({
  headingStyle: {
    marginTop: 16,
    paddingVertical: 16,
    paddingLeft: 16,
    backgroundColor: theme.colors.moreSeparator,
  },
  formContainer: {
    backgroundColor: theme.colors.white,
    paddingHorizontal: 24,
    paddingBottom: 24,
  },
  contentView: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  subContentView: {
    flex: 1,
    marginRight: 16,
  },
  flexOne: {
    flex: 1,
  },
  marginTop: {
    marginTop: 24,
  },
  inputFieldStyle: {
    height: 40,
  },
  dateStyle: {
    marginLeft: 0,
  },
  placeHolderStyle: {
    color: theme.colors.darkTint8,
  },
});
