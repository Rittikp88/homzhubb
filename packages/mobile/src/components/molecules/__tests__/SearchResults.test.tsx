import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchResults } from '@homzhub/mobile/src/components/molecules/SearchResults';
import { autocompleteMock } from '@homzhub/common/src/mocks/GooglePlacesMocks';

let props: any;
let wrapper: ShallowWrapper;

describe('SearchResults', () => {
  const createTestProps = (testProps: any): object => ({
    results: autocompleteMock,
    onResultPress: jest.fn(),
    listTitleStyle: {},
    ...testProps,
  });
  props = createTestProps({});

  it('should match snapshot', () => {
    wrapper = shallow(<SearchResults {...props} t={(key: string): string => key} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
