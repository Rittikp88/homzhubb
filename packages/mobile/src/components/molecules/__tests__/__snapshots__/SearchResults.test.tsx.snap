// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should match snapshot 1`] = `
<SearchResults
  i18n={Object {}}
  listTitleStyle={Object {}}
  onResultPress={[MockFunction]}
  results={
    Object {
      "predictions": Array [
        Object {
          "description": "Skyline Drive, Front Royal, VA, USA",
          "id": "5acbd52fbd6b0b3de0188098fe8db78a8c9da939",
          "matched_substrings": Array [
            Object {
              "length": 7,
              "offset": 0,
            },
          ],
          "place_id": "EiNTa3lsaW5lIERyaXZlLCBGcm9udCBSb3lhbCwgVkEsIFVTQSIuKiwKFAoSCYHx83sTSbSJEV1IwH3456fxEhQKEgnDiLgttMW1iRF4Tul0g-os2g",
          "reference": "EiNTa3lsaW5lIERyaXZlLCBGcm9udCBSb3lhbCwgVkEsIFVTQSIuKiwKFAoSCYHx83sTSbSJEV1IwH3456fxEhQKEgnDiLgttMW1iRF4Tul0g-os2g",
          "structured_formatting": Object {
            "main_text": "Skyline Drive",
            "main_text_matched_substrings": Array [
              Object {
                "length": 7,
                "offset": 0,
              },
            ],
            "secondary_text": "Front Royal, VA, USA",
          },
          "terms": Array [
            Object {
              "offset": 0,
              "value": "Skyline Drive",
            },
            Object {
              "offset": 15,
              "value": "Front Royal",
            },
            Object {
              "offset": 28,
              "value": "VA",
            },
            Object {
              "offset": 32,
              "value": "USA",
            },
          ],
          "types": Array [
            "route",
            "geocode",
          ],
        },
        Object {
          "description": "Skyline Gate Staging Area, Skyline Boulevard, Oakland, CA, USA",
          "id": "bd8d1f559490c3a1045c86258f0c4438d925632f",
          "matched_substrings": Array [
            Object {
              "length": 7,
              "offset": 0,
            },
          ],
          "place_id": "ChIJ2dcxm-WHj4ARCb3uXMSH7bo",
          "reference": "ChIJ2dcxm-WHj4ARCb3uXMSH7bo",
          "structured_formatting": Object {
            "main_text": "Skyline Gate Staging Area",
            "main_text_matched_substrings": Array [
              Object {
                "length": 7,
                "offset": 0,
              },
            ],
            "secondary_text": "Skyline Boulevard, Oakland, CA, USA",
          },
          "terms": Array [
            Object {
              "offset": 0,
              "value": "Skyline Gate Staging Area",
            },
            Object {
              "offset": 27,
              "value": "Skyline Boulevard",
            },
            Object {
              "offset": 46,
              "value": "Oakland",
            },
            Object {
              "offset": 55,
              "value": "CA",
            },
            Object {
              "offset": 59,
              "value": "USA",
            },
          ],
          "types": Array [
            "park",
            "point_of_interest",
            "establishment",
          ],
        },
        Object {
          "description": "Skyline Plaza Frankfurt, Europa-Allee, Frankfurt, Germany",
          "id": "f681e75dd3dd1401050f516b6b44805328a2582a",
          "matched_substrings": Array [
            Object {
              "length": 7,
              "offset": 0,
            },
          ],
          "place_id": "ChIJVVVVVVgJvUcRnLadHAeWTSA",
          "reference": "ChIJVVVVVVgJvUcRnLadHAeWTSA",
          "structured_formatting": Object {
            "main_text": "Skyline Plaza Frankfurt",
            "main_text_matched_substrings": Array [
              Object {
                "length": 7,
                "offset": 0,
              },
            ],
            "secondary_text": "Europa-Allee, Frankfurt, Germany",
          },
          "terms": Array [
            Object {
              "offset": 0,
              "value": "Skyline Plaza Frankfurt",
            },
            Object {
              "offset": 25,
              "value": "Europa-Allee",
            },
            Object {
              "offset": 39,
              "value": "Frankfurt",
            },
            Object {
              "offset": 50,
              "value": "Germany",
            },
          ],
          "types": Array [
            "shopping_mall",
            "point_of_interest",
            "establishment",
          ],
        },
        Object {
          "description": "Skyline Drive Scenic Roadway, South Terminus, Skyline Drive, Afton, VA, USA",
          "id": "626fe7f20feb9606114a25bb7e3a249b79e3d789",
          "matched_substrings": Array [
            Object {
              "length": 7,
              "offset": 0,
            },
          ],
          "place_id": "ChIJsw6Z2TNls4kRWcVwsI36BIQ",
          "reference": "ChIJsw6Z2TNls4kRWcVwsI36BIQ",
          "structured_formatting": Object {
            "main_text": "Skyline Drive Scenic Roadway, South Terminus",
            "main_text_matched_substrings": Array [
              Object {
                "length": 7,
                "offset": 0,
              },
            ],
            "secondary_text": "Skyline Drive, Afton, VA, USA",
          },
          "terms": Array [
            Object {
              "offset": 0,
              "value": "Skyline Drive Scenic Roadway, South Terminus",
            },
            Object {
              "offset": 46,
              "value": "Skyline Drive",
            },
            Object {
              "offset": 61,
              "value": "Afton",
            },
            Object {
              "offset": 68,
              "value": "VA",
            },
            Object {
              "offset": 72,
              "value": "USA",
            },
          ],
          "types": Array [
            "point_of_interest",
            "establishment",
          ],
        },
        Object {
          "description": "Skyline Drive In Blacktown, Cricketers Arms Road, Blacktown NSW, Australia",
          "id": "be7e42d10ebc6c1c5d66bf9aec364290834c9e57",
          "matched_substrings": Array [
            Object {
              "length": 7,
              "offset": 0,
            },
          ],
          "place_id": "ChIJVVCGitOZEmsR2m8B7wkTQYs",
          "reference": "ChIJVVCGitOZEmsR2m8B7wkTQYs",
          "structured_formatting": Object {
            "main_text": "Skyline Drive In Blacktown",
            "main_text_matched_substrings": Array [
              Object {
                "length": 7,
                "offset": 0,
              },
            ],
            "secondary_text": "Cricketers Arms Road, Blacktown NSW, Australia",
          },
          "terms": Array [
            Object {
              "offset": 0,
              "value": "Skyline Drive In Blacktown",
            },
            Object {
              "offset": 28,
              "value": "Cricketers Arms Road",
            },
            Object {
              "offset": 50,
              "value": "Blacktown",
            },
            Object {
              "offset": 60,
              "value": "NSW",
            },
            Object {
              "offset": 65,
              "value": "Australia",
            },
          ],
          "types": Array [
            "movie_theater",
            "point_of_interest",
            "establishment",
          ],
        },
      ],
      "status": "OK",
    }
  }
  t={[Function]}
  tReady={false}
/>
`;
